### ISSUE TITLE
🐛 Biome doesn't throw errors for files that can't handle

### ISSUE THREAD

**Issue by ematipico** [biomejs/biome#1247]
### Environment information

```block
main
```

### What happened?

Try to handle a file that Biome doesn't know how to handle.

I should print a diagnostic saying that it doesn't know how to handle a file. This is a regression that was introduced way back, and we didn't catch because we don't have a test for it

### Expected result

I should throw a diagnostic. Users should use `files.ignore` or `files.ignoreUnknown`

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct
